### PR TITLE
feat: Add multi input console type to clean up final questions

### PR DESCRIPTION
**Pull request by IAmJSD**
Cleaning up the final questions in the CLI was a suggestion by Adam. This does this by cleanly showing the other options in a form-like component.

full size
truncated first page
truncated second page

### CODE REVIEW

**Review by jimeh — COMMENTED** [review]
This is very impressive and cool. But I came across a few oddities/issues while using it:

- On larger terminals (360x100) there's a lot of flicker and split second duplication of the form whenever I type. This is not an issue on smaller terminals (100x40). I'm not sure if this a performance issue, or just a screen clear/reset bug.
- When resizing the terminal window the form does not adjust to the new size until pressing a key which causes a redraw of the form. Ideally it should redraw when the terminal is resized.
- Pressing TAB on the multi-input form just inserts a tab character, and brakes the box rendering. I kind of would have expected this to cycle between inputs.
- Using arrow keys to navigate between the input fields works fine, but the wrap around only works with the down arrow key, pressing up on the first input field does not select the last field.
- Using ctrl+p (previous-line/up) and ctrl+n (next-line/down) do not work in place of arrow keys, which I believe they should as most shells and terminal apps treat those key sequences the same as up/down arrow keys.
- Pressing Enter just submitted the form creating the VM without any user-feedback at all about what it's doing, making it seem like the CLI just silently exited. Maybe this could also be improved with a little piece of text instructing that pressing enter/return will submit the create request, or simply have a Create/Continue TUI button that needs to be navigated to?

And not directly related to this PR:

- Tag selection for a VM is somewhat counter-intuitive, as enter toggles selections, and ESC actually moves you on to the next step. I'm not sure what the best way might be to improve this. Maybe have a Continue item on the list of tags?
- On the searchable list/table where you pick VM package, OS, etc, arrow keys navigate up/down, but ctrl+p (previous-line, up) and ctrl+n (next-line, down) seem to insert characters which immediately eliminates all results.

---

And this brings me on to a somewhat bigger question with all of this fancy TUI stuff we're doing. As awesome as it is, we are hand-rolling it all from scratch right right now. I wonder if it might be worth the overhead/effort to switch to using a some form of TUI package—like [tcell](https://github.com/gdamore/tcell), [gocui](https://github.com/jroimartin/gocui), or [bubbletea](https://github.com/charmbracelet/bubbletea)—that'll hopefully give us things like user input/text boxes, windows, forms, progress bars, etc.

Apologies if this has already been extensively discussed/evaluated already, to my knowledge it hasn't yet. I'm mostly just worried that the further we go down this path of hand-rolling the TUI, the more I think we'll be re-inventing the wheel, rather than focusing on making the CLI itself do what we want. And I think it could lead to quite high maintenance overhead down the road too. What do you all think? /cc @JakeMakesStuff @strideynet @adamcooke

**Comment by JakeMakesStuff** [comment]
The issue with the current packages in my opinion is the following:

1) They don't support everything we need. Recent additions to Bubble Tea have got closer to what we need, but not close enough for the table list view with search for example.
2) If we were to combine packages to achieve what we need, it would make the codebase extremely complex. For example, in the virtual machine builder we may need to combine several console libraries.
3) A lot of the current libraries are not designed to be unit tested and have very low test coverage themselves. 

If we can overcome those issues I'm for switching to a library, but I don't know we can really.

**Comment by jimeh** [comment]
I'm not sure we need to fully mix and match and combine different TUI libraries, cause I agree that would be horrible :)

But from what I can tell, something like tcell might be a good foundation to still hand-roll stuff on top of, if it can handle things like drawing/re-drawing, auto-resizing boxes/elements, etc., allowing us to focus on what should be displayed, rather than how we make it display correctly. If we still have to implement things ourselves on top of that, I think that's fine. At the very least, I think some basic console drawing libraries might help make our lives easier, instead of manually handling all re-draws and updates to the screen.

That said, I haven't used any of the libraries myself that I suggest, they simple seem like the go-to solutions. tcell does look pretty promising, for example [fzf](https://github.com/junegunn/fzf) uses it.

And I just noticed there's [tview](https://github.com/rivo/tview/) built on top if tcell, which sounds promising, to quote the readme:

> Among these components are:
>
> - Input forms (include input/password fields, drop-down selections, checkboxes, and buttons)
> - Navigable multi-color text views
> - Sophisticated navigable table views
> - Flexible tree views
> - Selectable lists
> - Grid, Flexbox and page layouts
> - Modal message windows
> - An application wrapper

File: cmd/katapult/console/chunk_reader.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	chunks     []chunk
+	pendingErr error
+	pipe       io.Reader
+	stopLoop   uintptr
+	m          sync.Mutex
+}
+
+func (c *chunkReader) loop() {
+	// Used to stop racing before the start of the loop.
+	c.m.Unlock()
```
[the remark this answers is not in the document]
Comment (JakeMakesStuff) [reply cmd/katapult/console/chunk_reader.go:25]: Yeah it's locked in the new function. The reasoning is that it stops a potential race by ensuring the loop has started before data is read.

[the remark this answers is not in the document]
Comment (strideynet) [reply cmd/katapult/console/chunk_reader.go:25]: As in, prevents someone calling `flush()` ?

[the remark this answers is not in the document]
Comment (JakeMakesStuff) [reply cmd/katapult/console/chunk_reader.go:25]: yeah, I feel flush should wait until it's fully initialised tbh, I can change this behaviour if wanted though

[the remark this answers is not in the document]
Comment (strideynet) [reply cmd/katapult/console/chunk_reader.go:25]: Hmm, it just feels odd when the mutex isn't all in one place. Easy to foot-gun but as long as it's what you intended thats fine.

**Review by strideynet — APPROVED** [review]

**Pull request merged**